core/src/main/java/bio/terra/pearl/core/model/Versioned.java

```diff
@@ -1,7 +1,16 @@
 package bio.terra.pearl.core.model;
 
+/**
+ * Versioned is used for customer-editable documents.  All documents must have a stableId and version, and the stableID-version
+ * pair must be unique.  stableId-publishedVersion pairs must also be unique, but not every document is required to have a publishedVersion
+ *
+ * PublishedVersion is only intended as a cosmetic feature for users to "tag" documents.  All application logic and data fetching should be based on
+ * version.  this ensures that the application will operate the same in the sandbox environment as it does in live environments.
+ */
```

Comment:
We talked about source control analogues in person. Another Analogue I wanted to mention is Confluence URLs. For example, in `https://broadworkbench.atlassian.net/wiki/spaces/PEARL/pages/2842230785/Code+coverage`, `2842230785` is the stable document ID while `Code+coverage` is the title. The page title can be changed, thus changing the URL, and it will still load the same document. https://broadworkbench.atlassian.net/wiki/spaces/PEARL/pages/2842230785 always works to load the document, and Confluence doesn't even care if the title is or ever was correct, so https://broadworkbench.atlassian.net/wiki/spaces/PEARL/pages/2842230785/foo also works! In both cases, the browser is redirected to the document with the current title in the URL.